### DISCUSSION TITLE
Find! vs Lookup! in DInfo

### DISCUSSION THREAD

**Discussion by pamoroso** [Interlisp/medley#1338] [General]
What's the difference between the `Find!` and the `Lookup!` command in the free menu of DInfo? The LispUsers documentation only mentions the latter, which looks up in the index.

**Comment by pamoroso** [comment]
According to the [DInfo](https://github.com/Interlisp/medley/blob/de4e5f5ca427aba038730da8f0909a3979f4ffbf/lispusers/DINFO#L584) source, the documentation string of `Find!` in the free menu says "Perform a string search in the selected text of the current node".

**Comment by masinter** [comment]
"use the source, Luke" (for Star Wars fans). 

I think it was considered adequate documentation for the menu to explain itself in situ vs in a separate document, with HELPSYS and later DINFO transforming the document into online. 

One of the things missing in Medley's common lisp is handling of documentation strings in some places -- perhaps this was a CLtL 2 addition?   Now you can look up arglist and search in several sources on typein (including the Common Lisp HyperSpec )  using the "man" command or "?<enter>" during typein, but I think this could be also linked to s-edit's meta-H.

**Comment by rmkaplan** [comment]
In principle, the FIND command in the DINFO menu should just be an alternative to being able to type Meta-F as in other Tedit windows and in most operating system windows:  Find a string in this document.  

In which case the FIND in the menu could be removed, and the confusion would go away.

**Reply by pamoroso** [reply]
Now that I know what it does I think having `Find!` in the menu is a handy shortcut and a visual reminder of the feature.
